packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -152,12 +152,19 @@ export interface FieldProps {
 	 * @defaultValue If not specified, the key that is persisted is the property key that was specified in the schema.
 	 */
 	readonly key?: string;
+
 	/**
 	 * A default provider used for fields which were not provided any values.
 	 * @privateRemarks
 	 * We are using an erased type here, as we want to expose this API but `InsertableContent` and `NodeKeyManager` are not public.
 	 */
 	readonly defaultProvider?: DefaultProvider;
+
+	/**
+	 * Optional metadata to associate with the field.
+	 * @remarks Note: this metadata is not persisted in the document. It only exists at runtime.
```

Comment:
Generally, we compare runtime vs compile time. Given your strong typing, this provides both runtime and compile time information, so I don't think this is good phrasing.

Also, the defaultProvider above is even more runtime only that this is, so calling out the runtime onlyness of metadata is confusing.

Reply:
Ah, fair. I'll update the phrasing. Given that some aspects of the field props affect what gets persisted, and some don't, I think calling out which bucket they fall into is important. But the terminology needs improvement.